Alluvial.Tests/(Infrastructure)/Wait.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Alluvial.Tests
+{
+    public static class Wait
+    {
+        public static async Task Until(
+            Func<bool> until,
+            TimeSpan? pollInterval = null,
+            TimeSpan? timeout = null)
+        {

```

Comment:
If you want it fully async it's not entirely straightforward. For imperative tasks, async / await is a little more readable I think. 



> On Feb 8, 2015, at 6:25 PM, Nathan Black <notifications@github.com> wrote:
> 
> In Alluvial.Tests/(Infrastructure)/Wait.cs:
> 
> > @@ -0,0 +1,68 @@
> > +using System;
> > +using System.Diagnostics;
> > +using System.Threading.Tasks;
> > +
> > +namespace Alluvial.Tests
> > +{
> > +    public static class Wait
> > +    {
> > +        public static async Task Until(
> > +            Func<bool> until,
> > +            TimeSpan? pollInterval = null,
> > +            TimeSpan? timeout = null)
> > +        {
> This takes like 3 lines to do in Rx right?
> 
> —
> Reply to this email directly or view it on GitHub.
>